### PR TITLE
Move onnx-mlir version to 0.3.0

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Move `onnx-mlir` version to `0.3.0` from `0.2.0`. `1.0.0` was a mistake.

Signed-off-by: Whitney Tsang <whitneyt@ca.ibm.com>

### CODE REVIEW

**Comment by tungld** [comment]
@whitneywhtsang can we make the version printed out when invoking `onnx-mlir --version`? Currently, I only see LLVM version:
```bash
./Debug/bin/onnx-mlir --version
LLVM (http://llvm.org/):
  LLVM version 15.0.0git
  Optimized build with assertions.
  Default target: s390x-ibm-linux
  Host CPU: z14
  ```

**Comment by whitneywhtsang** [comment]
> can we make the version printed out when invoking onnx-mlir --version?

For sure, I am actually looking into that, which I will do in a separate PR.

**Comment by whitneywhtsang** [comment]
> > can we make the version printed out when invoking onnx-mlir --version?
> 
> For sure, I am actually looking into that, which I will do in a separate PR.

Created https://github.com/onnx/onnx-mlir/pull/1440.

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,11 @@ Releases are extensively tested, including the following steps.
 
 # Current releases
 
+## Prerelease 0.3.0
+
+This prerelease was cut on May TBD, 2022.
```
Comment (etiotto) [inline CHANGELOG.md:16]: I propose Fri May 20, 2022 as the date.

**Comment by whitneywhtsang** [comment]
@AlexandreEichenberger @etiotto I have set the date as May 20th, 2022.

**Pull request merged**